server/backend/test/steps/get-notifications.steps.ts

```diff
@@ -0,0 +1,180 @@
+import { INestApplication } from '@nestjs/common';
+import { defineFeature, loadFeature } from 'jest-cucumber';
+import request from 'supertest';
+import { Camera } from '../../src/cameras/camera.entity';
+import { Notification } from '../../src/cameras/notification.entity';
+import {
+  buildTestStack,
+  TestStack,
+  TEST_STACK_TIMEOUT,
+} from '../helpers/test-stack';
+import { EntityManager, EntityRepository } from '@mikro-orm/core';
+import { MikroORM } from '@mikro-orm/core';
+import { CamerasModule } from '../../src/cameras/cameras.module';
+import { Group } from '../../src/groups/group.entity';
+import { User } from '../../src/users/user.entity';
+import { createCameraWithKeyPair, getCameraAuthToken } from '../helpers/camera';
+
+const feature = loadFeature('test/features/get-notifications.feature');
+
+defineFeature(feature, (test) => {
+  let app: INestApplication;
+  let testStack: TestStack;
+  let cameraRepository: EntityRepository<Camera>;
+  let em: EntityManager;
+
+  beforeAll(async () => {
+    testStack = await buildTestStack({ imports: [CamerasModule] });
+
+    em = testStack.module.get<MikroORM>(MikroORM).em.fork();
+    cameraRepository = em.getRepository(Camera);
+
+    app = testStack.module.createNestApplication();
+    await app.init();
+  }, TEST_STACK_TIMEOUT);
+
+  afterAll(async () => {
+    await app.close();
+    await testStack.dbContainer.stop();
+  });
+
+  test('Using a valid ID with 2 notifications to get', ({
+    given,
+    and,
+    when,
+    then,
+  }) => {
+    let cameraA: Camera;
+    let getRes: request.Response;
+    let token: string;
+
+    given('I have a valid ID', async () => {
+      const { camera, keyPair } = createCameraWithKeyPair('Camera-A');
+      cameraA = camera;
+      cameraA.group = new Group('group');
+      cameraA.group.user = new User();
+      await cameraRepository.persistAndFlush(cameraA);
+      token = getCameraAuthToken(cameraA, keyPair.privateKey);
+    });
+    and('the ID has 2 notifications associated with it', async () => {
+      cameraA.notifications.add(new Notification());
+      cameraA.notifications.add(new Notification());
+      await cameraRepository.flush();
+    });
+    when('I request to get the notifications', async () => {
+      getRes = await request(app.getHttpServer())
+        .get(`/cameras/${cameraA.id}/notifications`)
+        .set('Authorization', token);
+    });
+    then('the request will go through', () => {
+      expect(getRes.status).toBe(200);
+    });
+    and('I will receive a Notification array of 2', async () => {
+      expect(getRes.header['content-type']).toMatch(/^application\/json/);
+      expect(getRes.body).toHaveLength(2);
+    });
```

Comment:
We should also verify that the data matches what we expect; we should take the user's notifications, turn them into the format we expect from this endpoint, and expect that the two match. I think some of the tests in the camera group steps do something like this, so refer to that for an example.